lib/rules/id-blacklist.js

```diff
@@ -141,6 +162,16 @@ module.exports = {
                         if (isInvalid(name)) {
                             report(node);
                         }
+
+                    // Report the last identifier in an ObjectPattern destructuring.
+                    } else if (
+                        !effectiveParent.computed &&
+                        effectiveParent.type === "Property" && effectiveParent.value === node.parent &&
+                        effectiveParent.parent.type === "ObjectPattern" && effectiveParent.parent.parent.left === effectiveParent.parent
```

Comment:
If I'm not missing something, this combination is sufficient to tell that `node.parent` (`MemberExpression` ) will be assigned a value:

`effectiveParent.type === "Property" && effectiveParent.value === node.parent &&
                        effectiveParent.parent.type === "ObjectPattern"`

so if we remove this:

`effectiveParent.parent.parent.left === effectiveParent.parent`

the rule will also (I believe correctly) report cases where `ObjectPattern` isn't LHS of an assignment:

```js
/* eslint id-blacklist: ["error", "bar"] */

({foo: { a: obj.bar }} = baz); // should be error?
```

Reply:
> so if we remove this:
effectiveParent.parent.parent.left === effectiveParent.parent

It's right! thanks!